Examples/ImageRegistrationMethod2/ImageRegistrationMethod2.cs

```diff
@@ -0,0 +1,105 @@
+/*=========================================================================
+*
+*  Copyright Insight Software Consortium
+*
+*  Licensed under the Apache License, Version 2.0 (the "License");
+*  you may not use this file except in compliance with the License.
+*  You may obtain a copy of the License at
+*
+*         http://www.apache.org/licenses/LICENSE-2.0.txt
+*
+*  Unless required by applicable law or agreed to in writing, software
+*  distributed under the License is distributed on an "AS IS" BASIS,
+*  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+*  See the License for the specific language governing permissions and
+*  limitations under the License.
+*
+*=========================================================================*/
+
+using System;
+
+
+namespace itk.simple.examples
+{
+
+
+    class IterationUpdate : Command
+    {
+
+        private ImageRegistrationMethod m_Method;
+
+        public IterationUpdate(ImageRegistrationMethod m)
+        {
+            m_Method = m;
+        }
+
+        public override void Execute()
+        {
+            VectorDouble pos = m_Method.GetOptimizerPosition();
+            Console.WriteLine("{0:3} = {1:10.5} : [{2}, {3}]",
+                              m_Method.GetOptimizerIteration(),
+                              m_Method.GetMetricValue(),
+                              pos[0], pos[1]);
+        }
+    }
+
+    class ImageRegistrationMethod1
```

Comment:
A bit strange that the class is named ...Method1 when the example number is 2. Possibly use a generic name ImageRegistrationMethodExample or specific name ImageRegistrationMethodExample2.